contract/models/account_analytic_account.py

```diff
@@ -259,38 +266,52 @@ def _create_invoice(self):
         invoice.compute_taxes()
         return invoice
 
+    @contextmanager
+    @api.multi
+    def logged_exception_savepoint(self, *args):
+        try:
+            with self.env.cr.savepoint() as ctx:
+                yield ctx
+        except:
+            _logger.exception(traceback.format_exc())
+            if args:
+                _logger.exception(*args)
+
     @api.multi
     def recurring_create_invoice(self):
         """Create invoices from contracts
 
         :return: invoices created
         """
         invoices = self.env['account.invoice']
+
+        error_msg_tmpl = ('Invoice creation failed for contract id %d'
+                          ' (traceback above).'
+                          ' Going on with other contracts, if any.')
         for contract in self:
-            ref_date = contract.recurring_next_date or fields.Date.today()
-            if (contract.date_start > ref_date or
-                    contract.date_end and contract.date_end < ref_date):
-                if self.env.context.get('cron'):
```

Comment:
Why are you removing this part of code? It should be here as you shouldn't raise any error (even masking it) when on cron mode, as this is a condition to check for not generating invoices, but to alert when you click interactively on the create invoice button on the contract.